force_bdss/data_sources/base_data_source_model.py

```diff
@@ -68,6 +68,50 @@ def _trigger_changes_slots(self, obj, name, new):
         if changes_slots:
             self.changes_slots = True
 
+    @on_trait_change('changes_slots')
+    def _update_slot_info(self):
```

Comment:
Apparently, this method implements not only the `slot_info` update, but also the `slot` signature. The signature is changed when the `len(slot_info) != len(attr)`. Maybe there is a traits-way to handle these two different changes?